File: android/app/src/main/java/com/legaltalk/india/data/model/Consultation.kt
package com.legaltalk.india.data.model

data class Consultation(
    val id: String,
    val clientId: String,
    val clientName: String,
    val lawyerId: String,
    val lawyerName: String,
    val type: String, // "chat", "voice", "video"
    val status: String, // "requested", "active", "completed", "cancelled"
    val startedAt: String? = null,
    val endedAt: String? = null,
    val totalMinutes: Int? = null,
    val ratePerMinute: Double,
    val totalCost: Double? = null,
    val lawyerReceipt: Double? = null,
    val platformCommission: Double? = null,
    val agoraChannelName: String? = null
)

data class ConsultationMessage(
    val id: String,
    val consultationId: String,
    val senderId: String,
    val senderName: String,
    val text: String,
    val timestamp: String
)

File: ios/LegalTalkIndia/Data/Model/Consultation.swift
import Foundation

struct Consultation: Codable, Identifiable {
    let id: String
    let clientId: String
    let clientName: String
    let lawyerId: String
    let lawyerName: String
    let type: String // "chat", "voice", "video"
    let status: String // "requested", "active", "completed", "cancelled"
    let startedAt: String?
    let endedAt: String?
    let totalMinutes: Int?
    let ratePerMinute: Double
    let totalCost: Double?
    let lawyerReceipt: Double?
    let platformCommission: Double?
    let agoraChannelName: String?
}

struct ConsultationMessage: Codable, Identifiable {
    let id: String
    let consultationId: String
    let senderId: String
    let senderName: String
    let text: String
    let timestamp: String
}
